src/plugins/ResizeReorder/ReorderHandle.js

```diff
@@ -0,0 +1,241 @@
+      PropTypes.string,
+      PropTypes.number
+    ]),
+  };
+
+  /**
+   * Instance of DOMMouseMoveTracker to capture mouse events
+   * @type {DOMMouseMoveTracker}
+   */
+  mouseMoveTracker = null;
+
+  /**
+   * Displacement of reorder cell while reordering
+   * @type {number}
+   */
+  distance = 0;
+
+  /**
+   * Set true while reordering
+   * @type {boolean}
+   */
+  animating = false;
+
+  /**
+   * Frame Id of requested animation frame
+   * @type {number | null}
+   */
+  frameId = null;
+
+  /**
+   * Initial position of scroll
+   * @type {number}
+   */
+  scrollStart = 0;
+
+  /**
+   * Left of column
+   * @type {number}
+   */
+  originalLeft = 0;
+
+  render() {
+    const style = {
+      height: this.props.height,
+    };
+
+    return (
+      <div
+        className={cx({
+          'fixedDataTableCellLayout/columnReorderContainer': true,
+          'fixedDataTableCellLayout/columnReorderContainer/active': false,
+        })}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.props.touchEnabled ? this.onMouseDown : null}
+        onTouchEnd={this.props.touchEnabled ? e => e.stopPropagation() : null}
+        onTouchMove={this.props.touchEnabled ? e => e.stopPropagation() : null}
+        style={style}>
+      </div>
+    );
+  }
+
+  /**
+   *
+   * @param {MouseEvent} event
+   */
+  onMouseDown = (event) => {
+    this.distance = 0;
+    this.animating = true;
+    this.scrollStart = getState().scrollX;
+    this.originalLeft = this.props.left;
+    this.initializeDOMMouseMoveTracker();
+    this.mouseMoveTracker.captureMouseMoves(event);
+    this.updateParentReorderingData({
+      isColumnReordering: true,
+      displacement: 0
+    });
+    this.frameId = requestAnimationFrame(this.updateState);
+  };
+
+  /**
+   * @param {number} deltaX
+   */
+  onMouseMove = (deltaX) => {
+    this.distance += deltaX * (this.props.isRTL ? -1 : 1);
+  };
+
+  onMouseUp = () => {
+    cancelAnimationFrame(this.frameId);
+    this.calculateColumnOrder();
+    this.animating = false;
+    this.frameId = null;
+    this.distance = 0;
+    this.mouseMoveTracker.releaseMouseMoves();
+    this.updateParentReorderingData({
+      isColumnReordering: false,
+      displacement: 0
+    });
+  };
+
+  initializeDOMMouseMoveTracker = () => {
+    this.mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+  };
+
+  updateParentReorderingData = (state) => {
+    this.props.updateParentReorderingData(state);
+  };
+
+  updateState = () => {
+    if (this.animating) {
+      this.frameId = requestAnimationFrame(this.updateState);
+    }
+    this.calculateDisplacementWithScroll();
+  };
+
+  calculateDisplacementWithScroll = () => {
+    const scrollStart = this.scrollStart;
+    let { isFixed, maxScrollX } = this.props;
+    let { scrollX } = getState();
+    let deltaX = this.distance;
+    if (!isFixed) {
```

Comment:
why do we need this check? Can't fixed columns also be reordered?

Reply:
Fixed columns can wither be on left or right and hence scrollbar can't be moved when we try to reorder them. This check skips updating scrollbar for fixed columns.